### PR TITLE
Added the cache for the getproblem function

### LINKED ISSUE #490: Cache page requests
Esp here - https://github.com/code100x/daily-code/blob/main/apps/web/components/utils.tsx#L11

Cache for 30 mins

In memory cache is fine

### PR DESCRIPTION
**Pull request by Ameerjafar**
Resolves #490  

### Checklist before requesting a review
- [ yes ] I have made by code review.
- [ yes ] I assure there is no similar/duplicate pull request regarding same issue

### CODE REVIEW

File: apps/web/components/utils.tsx [hunk 1]
```diff
@@ -21,13 +26,22 @@ export async function getProblem(problemId: string | null) {
...
     return problem;
   } catch (err) {
     return null;
   }
 }
 
 export async function getFirstProblemForTrack(trackId: string) {
+  const value = await cache.get("tracks", [trackId.toString()]);
+  if(value) {
+    console.log("successfully added");
```
Comment (siinghd) [inline apps/web/components/utils.tsx:41]: remove console.log

File: apps/web/components/utils.tsx [hunk 2]
```diff
@@ -161,6 +186,11 @@ export async function getTrack(trackId: string) {
 }
 
 export async function getAllTracks() {
+  const value = await cache.get('getAllTracks', []);
+  if(value) {
+    console.log("get successfully");
```
Comment (siinghd) [inline apps/web/components/utils.tsx:191]: remove console.log

File: apps/web/components/utils.tsx [hunk 3]
```diff
@@ -188,6 +218,9 @@ export async function getAllTracks() {
         createdAt: "asc",
       },
     });
+    console.log(tracks);
```
Comment (siinghd) [inline apps/web/components/utils.tsx:221]: remove console.log

File: apps/web/components/utils.tsx [hunk 4]
```diff
@@ -188,6 +218,9 @@ export async function getAllTracks() {
...
+    await cache.set('getAllTracks', [], tracks);
+    console.log("hello set successfully");
```
Comment (siinghd) [inline apps/web/components/utils.tsx:223]: remove console.log

**Review by siinghd — CHANGES_REQUESTED** [review]

File: apps/web/components/utils.tsx (hunk 1 above)
Comment (Ameerjafar) [reply apps/web/components/utils.tsx]: done sir

File: apps/web/components/utils.tsx (hunk 3 above)
Comment (Ameerjafar) [reply apps/web/components/utils.tsx]: done sir.

**Comment by Ameerjafar** [comment]
Any bounties😁😁😁? @hkirat

**Pull request merged**